src/Tools/ExternalAccess/Xaml/Completion/XamlCompletionKind.cs

```diff
@@ -0,0 +1,177 @@
+    MarkupExtensionClass,
+
+    /// <summary>
+    /// The completion represents the name of a MarkupExtension parameter.
+    /// </summary>
+    MarkupExtensionParameter,
+
+    /// <summary>
+    /// The completion represents the value of a MarkupExtension parameter.
+    /// </summary>
+    MarkupExtensionValue,
+
+    /// <summary>
+    /// The completion represents a type for an attached property in the Property Completion.
+    /// (i.e. "Grid.").
+    /// </summary>
+    Type,
+
+    /// <summary>
+    /// The completion represents a value for Property attribute in Styles. (These are direct DPs on TargetTypes).
+    /// (i.e. "Grid.Background").
+    /// </summary>
+    PropertyValue,
+
+    /// <summary>
+    /// The completion represents a value for Property attribute in Styles. (These are APs on types).
+    /// (i.e. "Grid.Row").
+    /// </summary>
+    AttachedPropertyValue,
+
+    /// <summary>
+    /// The completion represents a type  within for Property attribute in Styles. (These are the types that for APs).
+    /// (i.e. "Grid.Row").
+    /// </summary>
+    AttachedPropertyTypePrefix,
+
+    /// <summary>
+    /// The completion represents a local resource.
+    /// </summary>
+    LocalResource,
+
+    /// <summary>
+    /// The completion represents a system resource.
+    /// </summary>
+    SystemResource,
+
+    /// <summary>
+    /// The completion represents a property from the schema generated from a data source.
+    /// </summary>
+    DataBoundProperty,
+
+    /// <summary>
+    /// The completion represents the name of an element in the current scope.
+    /// </summary>
+    ElementName,
+
+    /// <summary>
+    /// The completion represents a namespace value. For instance, xmlns:local="Completion"
+    /// </summary>
+    NamespaceValue,
+
+    /// <summary>
+    /// The completion represents a condition value in a namespace. For instance, xmlns:local="namespace?Completion"
+    /// </summary>
+    ConditionValue,
+
+    /// <summary>
+    /// The completion represents a conditional argument value in a namespace. For instance, xmlns:local="namespace?Condition(Completion)"
+    /// </summary>
+    ConditionalArgument,
+
+    /// <summary>
+    /// A completion that cannot legally be used, but is shown for sake of user-education or
+    /// completeness. Example would be the phone Pivot control in a shared Mercury XAML file:
+    /// this type cannot be legally used in a shared context, but we want to show it as it
+    /// is a core phone type.
+    /// </summary>
+    Unusable,
+
+    /// <summary>
+    /// The completion represents #region for XAML
+    /// </summary>
+    RegionStart,
+
+    /// <summary>
+    /// The completion represents #endregion for XAML
+    /// </summary>
+    RegionEnd,
+
+    /// <summary>
+    /// The completion represents a snippet for XAML
+    /// </summary>
+    Snippet,
+
+    /// <summary>
+    /// The completion represents a method
+    /// </summary>
+    Method,
```

Comment:
Can we also add 
```
        /// <summary>
        /// The completion represents a color value
        /// </summary>
        Color,
```

Reply:
@etvorun This is one of those things that I'm considering moving out of Roslyn, so that we end up doing any XAML to LSP mapping in the VS repo.